src/net_processing.cpp

```diff
@@ -1634,19 +1645,25 @@ CTransactionRef static FindTxForGetData(CNode& peer, const uint256& txid, const
 
     auto txinfo = mempool.info(txid);
     if (txinfo.tx) {
-        // To protect privacy, do not answer getdata using the mempool when
-        // that TX couldn't have been INVed in reply to a MEMPOOL request,
-        // and it's more recent than UNCONDITIONAL_RELAY_DELAY.
+        // If a TX could have been INVed in reply to a MEMPOOL request,
+        // or is older than UNCONDITIONAL_RELAY_DELAY, permit the request
+        // unconditionally.
         if ((mempool_req.count() && txinfo.m_time <= mempool_req) || txinfo.m_time <= now - UNCONDITIONAL_RELAY_DELAY) {
-            return txinfo.tx;
+            return std::move(txinfo.tx);
         }
     }
 
     {
         LOCK(cs_main);
-        // Look up transaction in relay pool
-        auto mi = mapRelay.find(txid);
-        if (mi != mapRelay.end()) return mi->second;
+
+        // Otherwise, the transaction must have been relayed recently.
```

Comment:
announced recently, not relayed?

Reply:
same question about the commit name:
"Only respond to requests for recently relayed transactions "

Reply:
@naumenkogs Yeah, announced is clearer terminology. I've changed it in the variable name, comments, and commit message.